### DISCUSSION TITLE
Mixing of `.def()` and `value()` on `nb::enum_`s?

### DISCUSSION THREAD

**Discussion by nicholasjng** [wjakob/nanobind#113] [Q&A] [answered]
A quick observation while porting a pre-existing `enum_` with an associated operator from pybind:

```
#include "nanobind/operators.h"
#include "nanobind/nanobind.h"

enum Flags {
  a = 0,
  b = 1 << 0,
  c = 1 << 1,
};

Flags inline operator|(const Flags& LHS, const Flags& RHS) {
  return static_cast<Flags>(static_cast<int>(LHS) | static_cast<int>(RHS));
}

NB_MODULE(hello, m) {
  nb::enum_<Flags>(m, "Flags")
      .value("a", Flags::a)
      .value("b", Flags::b)
      .value("c", Flags::c)
      .export_values()
      .def(nb::self | nb::self);  // <- error: member reference base type 'void' is not a structure or union
};
```
This does not compile with nanobind, because `nb::enum_::export_values` returns `void`, whereas in pybind11 it returns a reference to the enum.

https://github.com/wjakob/nanobind/blob/a7d8dc08657ada946e174b849686dc969376ffc6/include/nanobind/nb_class.h#L467

Is this an intended break with pybind11, and if so, how could I define said operator on an `nb::enum` after exporting the values?

(PS: I found out about `nb::is_arithmetic()` while attempting to fix this , which would add that operator for me - but that returns the associated integer, while the custom operator would return an enum value, even if the resulting integer is not an existing value. My question is more about custom `def`s on enums in general.)

**Comment by wjakob** [comment, marked as answer]
This was unintended. Commit https://github.com/wjakob/nanobind/commit/714d17e71aa405c7633e0bd798a8bdb7b8916fa1 changes the signature to be consistent with pybind11.

**Reply by nicholasjng** [reply]
Awesome, thank you!
